Implement deleteFile graphql mutation (#5292).

francois2metz opened the pull request:
This fix #5292 by implementing and testing the removeFile mutation.

alexandrebodin commented on packages/strapi-plugin-upload/config/schema.graphql.js:
```diff
@@ -57,6 +56,15 @@ module.exports = {
       },
+      deleteFile: {
+        description: 'Delete one file',
+        resolverOf: 'plugins::upload.upload.deleteFile',
```
you are referencing a controller that doesn't exist :) it would have to be the destroy controller action :)

francois2metz replied:
Curious that the unit tests were ok. Fixed!

alexandrebodin commented on packages/strapi-plugin-upload/config/schema.graphql.js:
```diff
@@ -57,6 +56,15 @@ module.exports = {
+        description: 'Delete one file',
+        resolverOf: 'plugins::upload.upload.destroy',
+        resolver: async (obj, options, { context }) => {
+          const file = await strapi.plugins.upload.services.upload.fetch({ id: context.params.id });
```
You will just need to check the file exist before trying to remove it

alexandrebodin replied:
I think you need to add a test on the file and throw a notFound error

francois2metz replied:
I added a test, but for some reason there is no error. I'll investigate.

francois2metz replied:
I don't know if the test is relevant, but I fixed it at least!

alexandrebodin commented:
Hi @francois2metz just a small update before we can merge ;)

francois2metz commented:
@alexandrebodin I'll try to do it today. Thanks for the reminder ;).

francois2metz commented:
I updated the PR to check if the file exists, and added a commit to fix a typo. I can remove it from this branch if you don't want to mix the history.

alexandrebodin commented:
Thansk for this PR !
